### DISCUSSION TITLE
Finding the pattern of a backdoor

### DISCUSSION THREAD

**Discussion by schroe-ja** [Trusted-AI/adversarial-robustness-toolbox#1737] [Q&A] [answered]
Hi,

I try to find the backdoor pattern from the `add_pattern_bd()` in my poisoned images. I need this pattern as an image file. Do you have this image or can I generate it from the function? I use this function in `PoisoningAttackBackdoor(add_pattern_bd)`.

**Comment by beat-buesser** [comment, marked as answer]
Hi @EvilWatermelon I think the easiest approach would be to run `add_pattern_bd()` on an array and set `add_pattern_bd()` then the patterns should be in the 3x3 pixels in the bottom-right corner and you select these pixels from the returned array and save as image. Would this work for you?

**Reply by schroe-ja** [reply]
Yes, that could work. OpenCV can convert arrays to images and back. Thanks!
